src/vos/vos_internal.h

```diff
@@ -970,12 +974,19 @@ struct vos_rsrvd_scm {
 	struct pobj_action	*rs_actv;
 };
 
+D_CASSERT(offsetof(struct vos_rsrvd_scm, rs_actv_cnt) ==
+	  offsetof(struct dtx_rsrvd_scm, drs_cnt));
+D_CASSERT(offsetof(struct vos_rsrvd_scm, rs_actv_at) ==
+	  offsetof(struct dtx_rsrvd_scm, drs_at));
+D_CASSERT(offsetof(struct vos_rsrvd_scm, rs_actv) ==
+	  offsetof(struct dtx_rsrvd_scm, drs_ptr));
+D_CASSERT(sizeof(struct vos_rsrvd_scm) == sizeof(struct dtx_rsrvd_scm));
+
 umem_off_t
 vos_reserve_scm(struct vos_container *cont, struct vos_rsrvd_scm *rsrvd_scm,
 		daos_size_t size);
 int
-vos_publish_scm(struct vos_container *cont, struct vos_rsrvd_scm *rsrvd_scm,
-		bool publish);
+vos_publish_scm(struct vos_container *cont, void *data, bool publish);
```

Comment:
why changed it to 'void *'?

Reply:
That allows we use the buffer that is attached to DTX handle as the input parameter.